DSNP-Wallet/Source/Common/Subscriptions/MSASubscription.swift

```diff
@@ -0,0 +1,94 @@
+//
+//  MsaSubscription.swift
+//  DSNP-Wallet
+//
+//  Created by Ryan Sheh on 4/11/23.
+//
+
+import Foundation
+import SubstrateSdk
+import RobinHood
+
+class MsaSubscription {
+    let chainRegistry: ChainRegistryProtocol
+    let storageRequestFactory: StorageRequestFactoryProtocol
+    
+    init(chainRegistry: ChainRegistryProtocol,
+         storageRequestFactory: StorageRequestFactoryProtocol) {
+        self.chainRegistry = chainRegistry
+        self.storageRequestFactory = storageRequestFactory
+    }
+        
+    func getMsaFrom(user: User, completion: @escaping (UInt32)->()) {
+        let chainId = FrequencyChain.shared.id
+        guard let publicKey = user.publicKey?.rawData(),
+              let connection = chainRegistry.getConnection(for: chainId),
+              let runtimeService = chainRegistry.getRuntimeProvider(for: chainId) else { return }
+        
+        let factory = GetMsaFromPublicKeyOperationFactory(connection: connection,
+                                                          runtimeService: runtimeService,
+                                                          storageRequestFactory: storageRequestFactory)
+        
+        let operationManager = OperationManagerFacade.sharedManager
+        let operationWrapper = factory.fetchGetMsaOperationWrapper(from: publicKey)
+        
+        operationWrapper.targetOperation.completionBlock = {
+            do {
+                let msa = try operationWrapper.targetOperation.extractNoCancellableResultData()
+                completion(msa)
+            } catch {
+                print("\(error)")
+            }
+        }
+        
+        operationManager.enqueue(operations: operationWrapper.allOperations, in: .transient)
+    }
+}
+
+class GetMsaFromPublicKeyOperationFactory {
+    let connection: JSONRPCEngine
```

Comment:
maybe make these `private` if clients are just using the `fetchGetMsaOperationWrapper` func